src/Libraries/Microsoft.Extensions.AI.OpenAI/OpenAIAssistantClient.cs

```diff
@@ -0,0 +1,379 @@
+#pragma warning disable SA1108 // Block statements should not contain embedded comments
+
+namespace Microsoft.Extensions.AI;
+
+/// <summary>Represents an <see cref="IChatClient"/> for an OpenAI <see cref="OpenAIClient"/> or <see cref="OpenAI.Chat.ChatClient"/>.</summary>
+internal sealed class OpenAIAssistantClient : IChatClient
+{
+    /// <summary>Key used to access an assistant id from AdditionalProperties.</summary>
+    private const string AssistantIdKey = "AssistantId";
+
+    /// <summary>The underlying <see cref="AssistantClient" />.</summary>
+    private readonly AssistantClient _assistantClient;
+
+    /// <summary>The ID of the assistant to use.</summary>
+    private readonly string? _assistantId;
+
+    /// <summary>The thread ID to use if none is supplied in <see cref="ChatOptions.ConversationId"/>.</summary>
+    private readonly string? _threadId;
+
+    /// <summary>Initializes a new instance of the <see cref="OpenAIAssistantClient"/> class for the specified <see cref="AssistantClient"/>.</summary>
+    /// <param name="assistantClient">The underlying client.</param>
+    /// <param name="assistantId">The ID of the assistant to use.</param>
+    /// <param name="threadId">The ID of the thread to use.</param>
+    public OpenAIAssistantClient(AssistantClient assistantClient, string? assistantId, string? threadId)
+    {
+        _assistantClient = Throw.IfNull(assistantClient);
+
+        _assistantId = assistantId;
+        _threadId = threadId;
+
+        // https://github.com/openai/openai-dotnet/issues/215
+        // The endpoint isn't currently exposed, so use reflection to get at them, temporarily. Once packages
+        // implement the abstractions directly rather than providing adapters on top of the public APIs,
+        // the package can provide such implementations separate from what's exposed in the public API.
+        Uri providerUrl = typeof(AssistantClient).GetField("_endpoint", BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance)
+            ?.GetValue(assistantClient) as Uri ?? OpenAIChatClient.DefaultOpenAIEndpoint;
+
+        Metadata = new("openai", providerUrl);
+    }
+
+    /// <inheritdoc />
+    public ChatClientMetadata Metadata { get; }
+
+    /// <inheritdoc />
+    public object? GetService(Type serviceType, object? serviceKey = null)
+    {
+        _ = Throw.IfNull(serviceType);
+
+        return
+            serviceKey is not null ? null :
+            serviceType == typeof(AssistantClient) ? _assistantClient :
+            serviceType.IsInstanceOfType(this) ? this :
+            null;
+    }
+
+    /// <inheritdoc />
+    public Task<ChatCompletion> CompleteAsync(
+        IList<ChatMessage> chatMessages, ChatOptions? options = null, CancellationToken cancellationToken = default) =>
+        CompleteStreamingAsync(chatMessages, options, cancellationToken).ToChatCompletionAsync(coalesceContent: true, cancellationToken);
```

Comment:
Presumably we could change this to use the nonstreaming APIs in the future if we wanted, and wouldn't consider that a breaking change.

Just asking because historically there have been behavioral/capability differences between streaming and nonstreaming, leading to practical reasons why the nonstreaming calls need to use the nonstreaming APIs.